packages/workers/src/main.ts

```diff
@@ -1,24 +1,17 @@
-import { ChainName, EvmChain } from '@metamask-institutional/defi-adapters'
-import { logger, updateLogger } from './logger.js'
+import { EvmChain } from '@metamask-institutional/defi-adapters'
+import { logger } from './logger.js'
 import { runner } from './runner.js'
 
-if (!process.env.CHAIN_ID) {
-  logger.error('Chain ID is required')
+if (!process.env.CACHE_DATABASE_URL) {
+  logger.error('CACHE_DATABASE_URL is required')
   process.exit(1)
 }
 
-const chainId = Number.parseInt(process.env.CHAIN_ID, 10) as EvmChain
+logger.info('Starting workers')
 
-if (!Object.values(EvmChain).includes(chainId)) {
-  logger.error({ chainId: process.env.CHAIN_ID }, 'Invalid chain ID')
-  process.exit(1)
-}
-
-updateLogger(chainId, ChainName[chainId])
-
-if (!process.env.DATABASE_URL) {
-  logger.error('DATABASE_URL is required')
-  process.exit(1)
-}
-
-await runner(process.env.DATABASE_URL, chainId)
+await Promise.all(
+  Object.values(EvmChain).map(async (chainId) => {
+    const childLogger = logger.child({ chainId })
+    await runner(process.env.CACHE_DATABASE_URL!, chainId, childLogger)
+  }),
+)
```

Comment:
Instead of pm2, it just creates one promise for each chain (that should never finish).

The logger with the chainId needs to be created and passed down as a prop.